QA: Scheduler Test Fails in Bash

Running 'make test' in Ubuntu LTS 20.04 bash shell results in following failures. 

The issue might be related to shell environment. 

```
FAIL
FAIL	github.com/dagu-org/dagu/internal/digraph/scheduler	32.534s

=== FAIL: internal/digraph/scheduler TestNode/OutputTabWithDoubleQuotes (0.24s)
time=2025-01-29T20:25:01.910+07:00 level=ERROR source=/home/arky/Code/Golang/dagu/internal/logger/logger.go:218 msg="failed to get the DAG context"
    node_test.go:211: 
        	Error Trace:	/home/arky/Code/Golang/dagu/internal/digraph/scheduler/node_test.go:120
        	            				/home/arky/Code/Golang/dagu/internal/digraph/scheduler/node_test.go:211
        	Error:      	Not equal: 
        	            	expected: "OUTPUT=hello\tworld"
        	            	actual  : "OUTPUT=hello\\tworld"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-OUTPUT=hello	world
        	            	+OUTPUT=hello\tworld
        	Test:       	TestNode/OutputTabWithDoubleQuotes
        	Messages:   	output variable value mismatch
    --- FAIL: TestNode/OutputTabWithDoubleQuotes (0.24s)

=== FAIL: internal/digraph/scheduler TestNode/OutputTabWithMixedQuotes (0.30s)
time=2025-01-29T20:25:02.073+07:00 level=INFO source=/home/arky/Code/Golang/dagu/internal/logger/logger.go:229 msg="Step execution started" step=1
    node_test.go:216: 
        	Error Trace:	/home/arky/Code/Golang/dagu/internal/digraph/scheduler/node_test.go:120
        	            				/home/arky/Code/Golang/dagu/internal/digraph/scheduler/node_test.go:216
        	Error:      	Not equal: 
        	            	expected: "OUTPUT=hello\tworld"
        	            	actual  : "OUTPUT=hello\\tworld"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-OUTPUT=hello	world
        	            	+OUTPUT=hello\tworld
        	Test:       	TestNode/OutputTabWithMixedQuotes
        	Messages:   	output variable value mismatch
    --- FAIL: TestNode/OutputTabWithMixedQuotes (0.30s)

=== FAIL: internal/digraph/scheduler TestNode (6.69s)

=== FAIL: internal/digraph/scheduler TestScheduler/HandlingJSONWithSpecialChars (0.61s)
time=2025-01-29T20:25:27.790+07:00 level=INFO source=/home/arky/Code/Golang/dagu/internal/logger/logger.go:229 msg="Step execution started" step=1
time=2025-01-29T20:25:27.891+07:00 level=INFO source=/home/arky/Code/Golang/dagu/internal/logger/logger.go:229 msg="Step execution started" step=2
    scheduler_test.go:726: 
        	Error Trace:	/home/arky/Code/Golang/dagu/internal/digraph/scheduler/scheduler_test.go:726
        	Error:      	Not equal: 
        	            	expected: "RESULT=value"
        	            	actual  : "RESULT=${OUT.key}"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-RESULT=value
        	            	+RESULT=${OUT.key}
        	Test:       	TestScheduler/HandlingJSONWithSpecialChars
        	Messages:   	expected output "value", got "RESULT=${OUT.key}"
    --- FAIL: TestScheduler/HandlingJSONWithSpecialChars (0.61s)

=== FAIL: internal/digraph/scheduler TestScheduler (32.16s)

DONE 529 tests, 7 failures in 44.089s
make: *** [Makefile:118: test] Error 1



```

--- a/internal/test/setup.go
+++ b/internal/test/setup.go
@@ -5,6 +5,7 @@ import (
 	"context"
 	"fmt"
 	"os"
+	"os/exec"
 	"path/filepath"
 	"sync"
 	"syscall"
@@ -87,8 +88,19 @@ func Setup(t *testing.T, opts ...TestHelperOption) Helper {
 		opt(&helper)
 	}
 
+	// setup the default shell for reproducible result
+	setShell(t, "sh")
+
 	t.Cleanup(helper.Cleanup)
 	return helper
+}
+
+func setShell(t *testing.T, shell string) {
+	t.Helper()
+
+	shPath, err := exec.LookPath(shell)
+	require.NoError(t, err, "failed to find shell")
+	os.Setenv("SHELL", shPath)
 }
 
 // Helper provides test utilities and configuration

